src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -87,14 +91,49 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
 		function.getLibDirectory().ifPresent(d -> registerWithClassLoaderProvider(d));
 	}
 
-	private void registerWithClassLoaderProvider(File libDirectory) {
+	void registerWithClassLoaderProvider(File libDirectory) {
 		try {
-			classLoaderProvider.addDirectory(libDirectory);
+			if(SystemUtils.IS_JAVA_1_8) {
+				String workerLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
+				String javaReverseLibLoading = System.getenv(Constants.FUNCTIONS_WORKER_JAVA_LOAD_APP_LIBS);
+				File workerLib = new File(workerLibPath);
+
+				if(verifyLibrariesExist (workerLib, workerLibPath)) {
+					if(Util.isTrue(javaReverseLibLoading)) {
+						// load client app jars first.
+						classLoaderProvider.addDirectory(libDirectory);
+						classLoaderProvider.addDirectory(workerLib);
+					} else {
+						// Default load worker jars first.
+						classLoaderProvider.addDirectory(workerLib);
+						classLoaderProvider.addDirectory(libDirectory);
+					}
+				}
+			} else {
+				classLoaderProvider.addDirectory(libDirectory);
+			}
 		} catch (Exception ex) {
 			ExceptionUtils.rethrow(ex);
 		}
 	}
 
+	boolean verifyLibrariesExist (File workerLib, String workerLibPath) throws FileNotFoundException{
```

Comment:
this method never returns false. make this void. If calling this method does not throw , the it is ok.